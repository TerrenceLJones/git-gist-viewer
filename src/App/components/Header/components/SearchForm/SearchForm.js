import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';

const SearchForm = ({ pristine, submitForm, submitting }) => {
    return (
        <form onSubmit={submitForm}>
            <Field
                component='input'
                name='searchValue'
                placeholder='Search&hellip;'
                type='text'
            />
            <button type='submit' disabled={pristine || submitting}>Search</button>
        </form>
    );
};

export default SearchForm;