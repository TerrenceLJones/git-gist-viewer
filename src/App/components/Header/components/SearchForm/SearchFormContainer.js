import React from 'react';
import { compose, withHandlers } from 'recompose';
import { connect } from 'react-redux';
import { reduxForm } from 'redux-form';
import { withRouter } from 'react-router-dom';
import { parse } from 'query-string';
import _ from 'lodash';

import SearchForm from './SearchForm';

let SearchFormContainer = (props) => {
    return <SearchForm {...props} submitForm={props.submitForm} />;
}

const mapStateToProps = (state, ownProps) => {
    const queryString = _.get(ownProps, 'location.search');
    const searchValue = _.get(parse(queryString), 'q');

    return {
        initialValues: { searchValue }
    };
};

const mapDispatchToProps = null;

const enhance = compose(
    withRouter,
    connect(mapStateToProps, mapDispatchToProps),
    reduxForm({ form: 'headerSearchForm' }),
    withHandlers({
        submitForm: ({ handleSubmit, history }) => handleSubmit(({ searchValue }) => {
            history.push(`/search/?q=${searchValue}`);
        })
    })
);

export default enhance(SearchFormContainer);
