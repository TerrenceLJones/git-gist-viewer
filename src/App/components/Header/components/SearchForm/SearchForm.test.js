import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

it('renders without crashing', () => {
    expect(shallow(<SearchForm />)).toMatchSnapshot();
});
