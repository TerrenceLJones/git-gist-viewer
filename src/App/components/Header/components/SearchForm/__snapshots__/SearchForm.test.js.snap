// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Field
          component="input"
          name="searchValue"
          placeholder="Search…"
          type="text"
/>,
        <button
          disabled={undefined}
          type="submit"
>
          Search
</button>,
      ],
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": "input",
          "name": "searchValue",
          "placeholder": "Search…",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search",
          "disabled": undefined,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Search",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Field
            component="input"
            name="searchValue"
            placeholder="Search…"
            type="text"
/>,
          <button
            disabled={undefined}
            type="submit"
>
            Search
</button>,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": "input",
            "name": "searchValue",
            "placeholder": "Search…",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
            "disabled": undefined,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
